src/devices/sound/mas3507d.h

```diff
@@ -67,10 +67,15 @@ class mas3507d_device : public device_t, public device_sound_interface
 		CMD_CONTROL_WRITE = 0x6a
 	};
 
-	enum { IDLE, STARTED, NAK, ACK, ACK2 } i2c_bus_state;
-	enum { UNKNOWN, VALIDATED, WRONG } i2c_bus_address;
-	enum { UNDEFINED, CONTROL, DATA_READ, DATA_WRITE, BAD } i2c_subdest;
-	enum { CMD_BAD, CMD_RUN, CMD_READ_CTRL, CMD_WRITE_REG, CMD_WRITE_MEM, CMD_READ_REG, CMD_READ_MEM } i2c_command;
+	enum { IDLE = 0, STARTED, NAK, ACK, ACK2 };
+	enum { UNKNOWN = 0, VALIDATED, WRONG };
+	enum { UNDEFINED = 0, CONTROL, DATA_READ, DATA_WRITE, BAD };
+	enum { CMD_BAD = 0, CMD_RUN, CMD_READ_CTRL, CMD_WRITE_REG, CMD_WRITE_MEM, CMD_READ_REG, CMD_READ_MEM };
+
+	uint8_t i2c_bus_state;
+	uint8_t i2c_bus_address;
+	uint8_t i2c_subdest;
+	uint8_t i2c_command;
```

Comment:
For what it’s worth, you can make enumerated types that can be saved.  You declare it like:

```
enum i2c_bus_state_t : uint8_t { IDLE = 0, STARTED, NAK, ACK, ACK2 };
...
i2c_bus_state_t i2c_bus_state;
```

Then in your `.cpp` file you use a helper macro to declare the stuff allowing it to be saved:
```
ALLOW_SAVE_TYPE(mas3507d_device::i2c_bus_state_t)
```

(Remember to avoid usin names ending in `_t` at global scope because they’re reserved by POSIX.  They’re fine at class scope, though.)

It’s OK how it is now, just pointing out that you can keep the enum type information and still save things if you want to.

Reply:
This actually is how I wanted to do it but ran into errors about not being able to save the items in save states. I'll make the changes, as well as the ones to remove STRUCT_MEMBER.